### ISSUE TITLE
Request no-any rule to ignore rest args

### ISSUE THREAD

**Issue by jedmao** [palantir/tslint#4437]
### Feature request

**Is your feature request related to a problem? Please describe.**
Using the `no-any` rule, I get frustrated when _destructuring_ `...rest` args:
```ts
    public emit(eventName: string, ...rest: any[]) {
```
In this case, the `emit` method is going to call an internal listener with `listener(...rest)`. I don't care what the types are. I just want to pass-through the `...rest` args to the listener. I run into this issue quite frequently.

It would be great if I didn't have to scatter `// tslint:disable-next-line:no-any` everywhere I need `...rest` args.

**Describe the solution you'd like**
```json
    "no-any": [true, "ignore-rest-args"],
```

**Describe alternatives you've considered**
I considered using generics, but it's too limiting:
```ts
    public emit<T>(eventName: string, ...rest: T[]) {
```
That would force the `...rest` args to have all of the same type. If the first `...rest` arg was a `string`, it would force all the rest to be `string` as well.

**Additional context**
- [TypeScript 2.1 | Object Spread and Rest](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html#object-spread-and-rest)
- [EventEmitter GitHub Gist](https://gist.github.com/jedmao/6e5b701e24167a438e93fab3716c4a7a)

**Comment by JoshuaKGoldberg** [comment]
Hi @jedmao! TS 3 added variadic rest arg types, so in theory you should be able to work around the restriction of `...rest: T[]`:

```typescript
type Func<T extends any[]> = (...args: T) => void;

let listener: Func<[string, number]> = (one: string, two: number) => { /* ... */ };
```

Does that solve your use case?

**Comment by jedmao** [comment]
@JoshuaKGoldberg I'm not sure how that solves the problem, as it still uses `any[]` on your first line. Also, it requires knowledge of `[string, number]`, which I really don't have. A listener could truly have any signature.

**Comment by jedmao** [comment]
It looks like the following might work:
```ts
    public emit(eventName: string, ...rest: Array<unknown>) {
```

**Comment by jedmao** [comment]
I take that back. This is still an issue.

**Comment by adidahiya** [comment]
I agree, I think this rule should be lenient (by default) about `...rest` args

**Comment by vmk1vmk** [comment]
Hi, I am new to the open source community and this is my first PR. I thought this would be a good first issue to work on and I would love to receive some feedback.

**Comment by JoshuaKGoldberg** [comment]
Fixed super cleanly in #4581! 💪
